File: beginner/pagina02/1045/1045.js
const input = require('fs').readFileSync('/dev/stdin', 'utf8');
const lines = input.split(' ').map(Number);

let sortedArray = [...lines].sort((a, b) => b - a)


const Main = (arr) => {

    let [ a, b, c ] = arr

    let a2 = Math.pow(a, 2) 
    let bc2 = (Math.pow(b, 2) + Math.pow(c, 2))

    let response = []

    if(a >= (b + c)){
       response.push(`NAO FORMA TRIANGULO`) 

    } else {
        if(a2 === bc2) response.push(`TRIANGULO RETANGULO`)
        if(a2 > bc2) response.push(`TRIANGULO OBTUSANGULO`)
        if(a2 < bc2) response.push(`TRIANGULO ACUTANGULO`)
        if(a === b && b === c) response.push(`TRIANGULO EQUILATERO`)
        if(a !== b ? b === c : b !== c) response.push(`TRIANGULO ISOSCELES`)
    }
    
    console.log(response.join('\n'))
}

Main(sortedArray)

File: beginner/pagina02/1045/1045.lua
local function bubbleSort(arr)
    for i = 1, #arr - 1 do
        for j = 1, #arr - i do
            if arr[j] < arr[j + 1] then
                local temp = arr[j]
                arr[j] = arr[j + 1]
                arr[j + 1] = temp
            end
        end
    end
    return arr
end

local x = io.read("*n")
local y = io.read("*n")
local z = io.read("*n")

local arr = bubbleSort({x, y, z})
local a, b, c = arr[1], arr[2], arr[3]

local a2 = a * a
local b2 = b * b
local c2 = c * c

if a >= b + c then
    print("NAO FORMA TRIANGULO")
else
    if a2 == b2 + c2 then
        print("TRIANGULO RETANGULO")
    end
    if a2 > b2 + c2 then
        print("TRIANGULO OBTUSANGULO")
    end
    if a2 < b2 + c2 then
        print("TRIANGULO ACUTANGULO")
    end
    if a == b and b == c then
        print("TRIANGULO EQUILATERO")
    end
    if (a == b and b ~= c) or (a == c and a ~= b) or (b == c and b ~= a) then
        print("TRIANGULO ISOSCELES")
    end
end